flytekit/core/task.py

```diff
@@ -132,9 +133,9 @@ def task(
 
 @overload
 def task(
-    _task_function: Callable[P, FuncOut],
+    _task_function: Callable[..., FuncOut],
     task_config: Optional[T] = ...,
-    cache: bool = ...,
+    cache: Union[bool, CachePolicy, AutoCache] = ...,
```

Comment:
this should be `Union[bool, AutoCache]`, since `CachePolicy` is just an implementation of the `AutoCache` protocol.